include/llama.h

```diff
@@ -256,6 +259,18 @@ extern "C" {
         int8_t       *  logits; // TODO: rename this to "output"
     } llama_batch;
 
+    // Input data for llama_decode / llama_encode
+    // It can contain text tokens and embeddings for one or many sequences
+    struct llama_batch_ext;
+
+    struct llama_batch_ext_token_info {
+        llama_token    token;
+        llama_pos      pos;
+        int32_t        n_seq_id;
+        llama_seq_id * seq_id;
+        int8_t         logits;
+    };
```

Comment:
This might not be very future-proof. Mixed-modality batches would have tokens, embeddings and tensors mixed together in the same batch. So calling `llama_batch_ext_get_token_info(batch, i);` is not always well-defined because it might not be a token at position `i`.

Maybe we can postpone this "token_info" API. I think all usages in the examples that require to read back info from the batch can be implemented in the example code without relying on the API. This way we can focus only on implementing only the API for creating batches and adding data to them. Later on when we have a better idea of the implementation, we can add a helper API to get info back from the batches.

Reply:
Yes I agree. Furthermore, this API requires doing a copy, so it won't be the best for performance. It's better to remove this API for now.

> I think all usages in the examples that require to read back info from the batch can be implemented in the example code without relying on the API.

This kind of logic is currently being used inside `llama-server`, not sure it appears on any other examples. I think I can make a thin wrapper for `llama_batch_ext` inside the example code. Feel free to tell me if you have a better idea.

Reply:
This API is removed in https://github.com/ggml-org/llama.cpp/pull/11875/commits/1d6ba97789ed71a16b2e0b1d6f89dfda2499ac44 , a new `server_batch` wrapper is added to manage token logits placement in the batch